### PR TITLE
Fix some market topics not showing

### PR DESCRIPTION
**Pull request by Calvin-LL**
Because of the `line-clamp-1` class in the `ContractTopicBreadcrumbs` component, when a market has many topics, some are hidden with ellipsis. This PR makes this component display all topics.

This problem came from [this market](https://manifold.markets/CollectedOverSpread/will-manifold-allow-users-to-view-a).

#### Before

Screenshot 2024-01-09 at 14 33 44

#### After

Screenshot 2024-01-09 at 14 34 19

### CODE REVIEW

**Review by bgila — APPROVED** [review]
didn't QA but I believe

**Comment by jahooma** [comment]
The line clamp is intentional. But I can see it being a bit restrictive. How about line-clamp-2 as a compromise?

If a market has 99 topics, we don't want to fill the entire top of the page with all of those topics

**Comment by jahooma** [comment]
Or actually, it should show just 1 line, and then you can click a '...' button to show them all

**Comment by jahooma** [comment]
Do you want to try to make that change? If not, then feel free to close the PR.

**Comment by Calvin-LL** [comment]
> Do you want to try to make that change? If not, then feel free to close the PR.

working on it

**Comment by Calvin-LL** [comment]
Sorry about the pushes, can't figure out how to run it locally without `GOOGLE_APPLICATION_CREDENTIALS_DEV`. Haven't worked with firebase for a few years now so not quite sure what's going on. So kinda just relying on vercel previews to test.

I've added a show more button. Any thoughts?

#### Closed

Screenshot 2024-01-09 at 17 20 43

#### Open

Screenshot 2024-01-09 at 17 21 45

**Comment by bgila** [comment]
hey sorry for blocking you. we can't give people those creds because they reveal user emails

**Comment by bgila** [comment]
and by blocking, i meant "not approving the deploys quickly enough"

File: web/components/contract/market-topics.tsx [hunk 1]
```diff
@@ -48,26 +49,55 @@ const ContractTopicBreadcrumbs = (props: {
 }) => {
   const { contract, topics } = props
 
+  const spanRef = useRef<HTMLSpanElement>(null)
+  const [isClamped, setClamped] = useState(false)
```
Comment (bgila) [inline web/components/contract/market-topics.tsx:53]:
```suggestion
  const [isClamped, setClamped] = useState(true)
```

Comment (Calvin-LL) [reply web/components/contract/market-topics.tsx]: You're right, also `handleResize` should be called once before `addEventListener` to set the initial value. I've pushed a commit to fix that.

**Comment by bgila** [comment]
@Calvin-LL sorry dude it's not quite working. Please edit the js manually on the client side, debug it, and we'll merge it when it works. DM me on discord if i don't respond

**Comment by Calvin-LL** [comment]
> @Calvin-LL sorry dude it's not quite working. Please edit the js manually on the client side, debug it, and we'll merge it when it works. DM me on discord if i don't respond

I think I the commit I just pushed would fix it. Sorry it's hard to debug on my end. ~~I can't find a question with way too many topics.~~ Nvm I can just make the window tiny.

**Comment by Calvin-LL** [comment]
@bgila looks like it works now! Try this market in vercel prod preview: https://prod-50fc7yx03-mantic.vercel.app/CodeandSolder/will-trump-debate-chris-christie-be

#### Closed



#### Opened

**Comment by bgila** [comment]
thank you! this is a very generous thing for you to have done for us and i appreciate it

**Pull request merged**